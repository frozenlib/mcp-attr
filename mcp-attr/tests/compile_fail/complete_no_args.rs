use mcp_attr::{server::{mcp_server, McpServer, complete_fn}, Result};

struct TestServer;

#[complete_fn]
async fn complete_no_args() -> Result<Vec<String>> {
    Ok(vec!["test".to_string()])
}

#[mcp_server]
impl McpServer for TestServer {
    #[prompt]
    async fn test_prompt(&self, #[complete(complete_no_args)] msg: String) -> Result<String> {
        Ok(msg)
    }
}

fn main() {}