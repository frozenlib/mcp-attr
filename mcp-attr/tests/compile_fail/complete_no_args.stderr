error[E0061]: this function takes 0 arguments but 2 arguments were supplied
 --> tests/compile_fail/complete_no_args.rs:6:10
  |
5 | #[complete_fn]
  | --------------
  | |
  | unexpected argument #1 of type `&std::string::String`
  | unexpected argument #2 of type `&mcp_attr::server::RequestContext`
6 | async fn complete_no_args() -> Result<Vec<String>> {
  |          ^^^^^^^^^^^^^^^^
  |
note: function defined here
 --> tests/compile_fail/complete_no_args.rs:6:10
  |
6 | async fn complete_no_args() -> Result<Vec<String>> {
  |          ^^^^^^^^^^^^^^^^
